src/txmempool.cpp

```diff
@@ -1027,6 +1034,161 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
     return memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 15 * sizeof(void*)) * mapTx.size() + memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) + memusage::DynamicUsage(vTxHashes) + cachedInnerUsage;
 }
 
+bool CTxMemPool::DumpMempool(FopenFn mockable_fopen_function, bool skip_file_commit) const
+{
+    auto start = GetSystemTime<std::chrono::microseconds>();
+
+    std::map<uint256, CAmount> mapDeltas;
+    std::vector<TxMempoolInfo> vinfo;
+    std::set<uint256> unbroadcast_txids;
+
+    static Mutex dump_mutex;
+    LOCK(dump_mutex);
+
+    {
+        LOCK(cs);
+        for (const auto &i : mapDeltas) {
+            mapDeltas[i.first] = i.second;
+        }
+        vinfo = infoAll();
+        unbroadcast_txids = GetUnbroadcastTxs();
+    }
+
+    auto mid = GetSystemTime<std::chrono::microseconds>();
```

Comment:
When touching this line, it could make sense to make it a behaviour change and use steady clock, see for example #25456.